src/panels/calendar/dialog-calendar-event-editor.ts

```diff
@@ -259,6 +260,11 @@ class DialogCalendarEventEditor extends LitElement {
     this._dtstart = new Date(
       ev.detail.value + "T" + this._dtstart!.toISOString().split("T")[1]
     );
+
+    // Prevent that the end date can be before the start date
```

Comment:
This makes sense as it means the end date is now updated so that it won't be before the start date. However, the end time can still be before the start time with this change.

e.g. if you start with: 
Start: 2022-12-01 8AM
End: 2022-12-02 6AM

Then update the start date to 2022-12-02, this will result in:
Start: 2022-12-02 8AM
End: 2022-12-02 6AM

I believe a common way to handle this is by also computing a duration, then anytime the start time is updated add the duration to re-compute the end time. Then the user manually adjusts the end time and its never possible to have backwards durations.

What do you think about that?

Reply:
Agreed. I also extended the "start time changed" check to move the end date to the next day if required (to ensure we keep the original event duration and now correctly cross over the date/midnight).

So the logic is now:
* Start date changed to after end date => auto adjustment
* Start time changed to after end time => auto adjustment
* End date changed to before start date => date popup does not allow that already
* End time changed to before start time => no auto adjustment, but will be detected when trying to save the event (toast shown to user)